source/server-discovery-and-monitoring/tests/integration/pool-cleared-error.yml

```diff
@@ -0,0 +1,197 @@
+# Test SDAM error handling.
+runOn:
+    # This test requires retryable writes and failCommand appName.
+    - minServerVersion: "4.4"
+      topology: ["replicaset", "sharded"]
+
+database_name: &database_name "sdam-tests"
+collection_name: &collection_name "pool-cleared-error"
+
+data: []
+
+tests:
+  - description: PoolClearedError does not mark server unknown
+    clientOptions:
+      retryWrites: true
+      maxPoolSize: 1
+      appname: poolClearedErrorTest
+    operations:
+      # Perform an operation to ensure the node is discovered.
+      - name: insertOne
+        object: collection
+        arguments:
+          document:
+            _id: 1
+      # Configure the next insert to fail with a network error which will
+      # clear the pool leaving it paused until the server is rediscovered.
+      - name: configureFailPoint
+        object: testRunner
+        arguments:
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+                failCommands: ["insert"]
+                closeConnection: true
+                appName: poolClearedErrorTest
+      # Start threads.
+      - name: startThread
+        object: testRunner
+        arguments:
+          name: thread1
+      - name: startThread
+        object: testRunner
+        arguments:
+          name: thread2
+      - name: startThread
+        object: testRunner
+        arguments:
+          name: thread3
+      - name: startThread
+        object: testRunner
+        arguments:
+          name: thread4
+      - name: startThread
+        object: testRunner
+        arguments:
+          name: thread5
+      - name: startThread
+        object: testRunner
+        arguments:
+          name: thread6
+      # Perform concurrent insert operations. The first one to execute will
+      # fail with a network error, mark the server Unknown, clear the pool,
+      # and retry.
```

Comment:
This test is covering the DRIVERS-1483 / DRIVERS-746 retry behavior, but the retryability specs haven't yet been updated to account for pre-handshake errors yet. Should we tweak this test to not have any dependency on retry behavior / skip it until those have been done?

Reply:
No this is not testing DRIVERS-746 because in this case the handshake succeeds and it's the insert command that fails. As for DRIVERS-1483, I suppose this test is very similar. Still I think we should keep this test and also complete DRIVERS-1483. I think the spec work for DRIVERS-746 can follow after.

Reply:
The only reason I bundle the two tickets together is that their implementation / spec work would be basically identical except for differing tests. That being said, we can probably get spec work for at least DRIVERS-1483 done before any drivers have to implement the changes for this ticket, so I'm okay with leaving the test as-is then.